search/src/search.rs

```diff
@@ -181,37 +190,63 @@ impl Search {
         Ok(self.highlight(snippet))
     }
 
-    fn get_fuzzy_preview(&self, term: &str, distance: &u8, body: String) -> Result<String> {
-        let pindex: HashMap<&str, Vec<(u32, u32)>> = HashMap::from([
-            ("flooding", vec![(2, 10)])
-        ]);
-        let (matched_term, start, end) = self.get_fuzzy_match(term, distance, pindex);
+    fn get_fuzzy_preview(
+        &self,
+        path: &str,
+        term: &str,
+        distance: &u8,
+        body: &str,
+    ) -> Result<String> {
+        let pindex: PageIndex = self
+            .index
+            .page_index(path)
+            .map_err(|_| SearchError("".to_string()))?;
+        let (matched_term, start, end) = self
+            .get_fuzzy_match(term, distance, pindex)
+            .map_err(|_| SearchError("".to_string()))?;
         // Another safe way to get substrings using char_indices
-        let start = body.char_indices().nth(start as usize).unwrap_or((0, ' ')).0;
-        let end = body.char_indices().nth(end as usize).unwrap_or((body.len()-1, ' ')).0;
-        let substring = &body[start..end];
-        Ok(substring.to_string().replace(&matched_term, &format!("**{}**", term)))
+        let start = body
+            .char_indices()
+            .nth(start.saturating_sub(20) as usize)
+            .unwrap_or((0, ' '))
+            .0;
+        let end = body
+            .char_indices()
+            .nth((end + 20) as usize)
+            .unwrap_or((body.len() - 1, ' '))
+            .0;
+        let substring = &format!("...{}...", &body[start..end]);
+        let substring = substring
+            .to_string()
+            .replace(&matched_term, &format!("**{}**", matched_term));
+        Ok(substring.replace('\n', " "))
     }
 
-    fn get_fuzzy_match(&self, term: &str, distance: &u8, pindex: HashMap<&str, Vec<(u32, u32)>>) -> (String, u32, u32) {
+    fn get_fuzzy_match(
+        &self,
+        term: &str,
+        distance: &u8,
```

Comment:
If I get it correctly, this is the maximum allowed distance for a fuzzy match, right? If so, let's rename the argument to make the code more comprehensible.

Reply:
I would stick with distance, since it is the standard term related to levensthein*distance*, what do you think?